### PR TITLE
[docs] Improvements to the FAQ.

### PR DESCRIPTION
**Pull request by ChrisCummins**
This restructures the HTML documentation's FAQ section and adds new questions.

Some tips for common debugging:                                                       
                                                                                                       
   - Why does my environment’s step() function return “done”?                                          
   - How do I debug crashes or errors?                                                                 
                                                                                                       
and for common development use-cases:                                                                  
                                                                                                       
   - I want to modify a CompilerGym environment, where do I start?                                                                                                                                    
   - I want to add a new CompilerGym environment, where do I start?                                    
                                                                                                       
This replaces the previous question about adding a new observation / reward signal that was specific to the LLVM environment.  

CleanShot 2022-03-01 at 19 58 40@2x

The goal is to provide some tips for common debugging scenarios, like #585 and #582.

### CODE REVIEW

File: docs/source/faq.rst [hunk 1]
```diff
@@ -65,25 +193,3 @@ stability. If you would like to build from source but do not require the
...
-<compiler_gym.views.RewardView.add_space>`.
-
-If you require modifying the underlying compiler service implementation, fork
-this project and build it from source (see `installation
-<https://github.com/facebookresearch/CompilerGym/blob/development/INSTALL.md>`_).
-Then modify the service implementation for the compiler that you are interested
-in. The service codebase is located at
-:code:`compiler_gym/envs/$COMPILER/service`, where :code:`$COMPILER` is the name
-of the compiler service you would wish to modify, e.g. llvm. Once done, send us
-a pull request!
```
Comment (mostafaelhoushi) [inline docs/source/faq.rst:89]: Why is that removed? Is it no longer valid?

**Review by mostafaelhoushi — APPROVED** [review]
LGTM

File: docs/source/faq.rst (hunk 1 above)
Comment (ChrisCummins) [reply docs/source/faq.rst:89]: I merged it into a more general "I want to modify a CompilerGym environment, where do I start?", which covers other kinds of changes
